vars/cloudFoundryDeploy.groovy

```diff
@@ -145,11 +148,30 @@ def deployCfNative (config) {
             export HOME=${config.dockerWorkspace}
             cf login -u \"${username}\" -p '${password}' -a ${config.cloudFoundry.apiEndpoint} -o \"${config.cloudFoundry.org}\" -s \"${config.cloudFoundry.space}\"
             cf plugins
-            cf ${deployCommand} ${config.cloudFoundry.appName?:''} ${config.deployType == 'blue-green'?'--delete-old-apps':''} -f '${config.cloudFoundry.manifest}' ${config.smokeTest}"""
+            cf ${deployCommand} ${config.cloudFoundry.appName ?: ''} ${blueGreenDeployOptions} -f '${config.cloudFoundry.manifest}' ${config.smokeTest}
+            ${(config.keepOldInstance && config.deployType == 'blue-green') ? "cf stop ${config.cloudFoundry.appName}-old" : ''}
```

Comment:
Maybe moving the condition into a private method would improve readability

Reply:
Totally agree - and done